### PR TITLE
[SDK-566] Change ApiService module tests

### PR DESCRIPTION
**Pull request by bitPogo**
## Description

This adds a rewritten version of the ApiService module tests. This is part of [SDK-566](https://gesundheitscloud.atlassian.net/browse/SDK-566) and depends on #131.

## Motivation and Context

Currently the ApiService module tests are not working and being in bad shape, so they needed maintenance.

## How is it being implemented?

It adds a fresh written module tests for the ApiService.

## How Has This Been Tested?

This is about tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the changelog accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: sdk-core/src/test/java/care/data4life/sdk/network/ApiServiceModuleTest.kt [hunk 1]
```diff
@@ -0,0 +1,725 @@
...
+        assertNull(request.headers[HEADER_ALIAS])
+        assertEquals(
+            actual = request.headers[HEADER_AUTHORIZATION],
+            expected = "Bearer $authToken"
+        )
+        assertEquals(
+            actual = request.method,
+            expected = "DELETE"
+        )
+    }
```
Comment (bitPogo) [inline sdk-core/src/test/java/care/data4life/sdk/network/ApiServiceModuleTest.kt:596]: I just want make obvious, that this should work, but it does not. I even not see, why `deleteDocument` has to be different then `deleteRecord` in terms of returning a Completable. Can we align `deleteDocument` and `deleteRecord` so they working similar?

File: sdk-core/src/test/java/care/data4life/sdk/network/ApiServiceModuleTest.kt [hunk 2]
```diff
@@ -0,0 +1,725 @@
...
+
+    data class TestConnection(
+        override val isConnected: Boolean
+    ) : NetworkConnectivityService
+
+    @Before
+    fun setUp() {
+        server = MockWebServer()
+
+        every { env.getApiBaseURL(any()) } returns "https://www.i-hate-fat-constructors.com"
```
Comment (wmontwe) [inline sdk-core/src/test/java/care/data4life/sdk/network/ApiServiceModuleTest.kt:74]: this should be example.com

**Review by wmontwe — CHANGES_REQUESTED** [review]

File: sdk-core/src/test/java/care/data4life/sdk/network/ApiServiceModuleTest.kt (hunk 2 above)
Comment (bitPogo) [reply sdk-core/src/test/java/care/data4life/sdk/network/ApiServiceModuleTest.kt]: done

**Review by wmontwe — APPROVED** [review]
👍

**Pull request merged**